framework/source/class/qx/bom/request/Xhr.js

```diff
@@ -1062,7 +1062,7 @@ qx.Bootstrap.define("qx.bom.request.Xhr",
       if (this._getProtocol() === "file:") {
         error = !this.responseText;
       } else {
-        error = !this.statusText && this.status !== 204;
+        error = this.status === 0;
```

Comment:
I think that this is correct; `status === 0` is because the communication failed - eg TCP timeout rather than the server reporting "500 blah".  Unless a non-conforming client is masking a TCP failure as something else I'd imagine it would be fine.  The code previously was inferring failure from no `statusText`, except where it was `status == 204`because that routinely sends no `statusText`.